apps/src/code-studio/progress.js

```diff
@@ -175,6 +175,26 @@ progress.renderCourseProgress = function(scriptData) {
 
   const mountPoint = document.createElement('div');
   $('.user-stats-block').prepend(mountPoint);
+
+  const calendarLessonsOverview = scriptData.showCalendar
```

Comment:
I think it might be better to do this on the server side. I would suggest creating a summarize_lesson_for_calendar method in lesson.rb. Then in script summarize you would have an entry that is ` calendarLessons: lessons.map(&:summarize_lesson_for_calendar)`.  This would allow you to pass that calendarLessons straight into a prop from ScriptOverview